### PR TITLE
Update SimplifyInference documentation

### LINKED ISSUE #6852: [Relay][Bug] nn.batch_norm not being simplified by SimplifyInference
cc @hypercubestart, who discovered this oddity

```python
r = """
#[version = "0.0.5"]
def @main(%data: Tensor[(1, 32, 32, 16), float32], %bn_gamma: Tensor[(16), float32], %bn_beta: Tensor[(16), float32], %bn_mean: Tensor[(16), float32], %bn_var: Tensor[(16), float32]) -> (Tensor[(1, 32, 32, 16), float32], Tensor[(16), float32], Tensor[(16), float32]) {
    nn.batch_norm(%data, %bn_gamma, %bn_beta, %bn_mean, %bn_var, axis=3) /* ty=(Tensor[(1, 32, 32, 16), float32], Tensor[(16), float32], Tensor[(16), float32]) */
}
"""
mod = tvm.parser.parse(r)
mod = relay.transform.SimplifyInference()(mod)
print(mod.astext())
```
the result is not what we expected, though, as the batch norm doesn’t get simplified:
```
#[version = "0.0.5"]
def @main(%data: Tensor[(1, 32, 32, 16), float32], %bn_gamma: Tensor[(16), float32], %bn_beta: Tensor[(16), float32], %bn_mean: Tensor[(16), float32], %bn_var: Tensor[(16), float32]) -> (Tensor[(1, 32, 32, 16), float32], Tensor[(16), float32], Tensor[(16), float32]) {
  nn.batch_norm(%data, %bn_gamma, %bn_beta, %bn_mean, %bn_var, axis=3) /* ty=(Tensor[(1, 32, 32, 16), float32], Tensor[(16), float32], Tensor[(16), float32]) */
}
```
is that to be expected? are we doing something dumb here?

### PR DESCRIPTION
**Pull request by gussmith23**
Closes #6852

### CODE REVIEW

**Review by comaniac — APPROVED** [review]
LGTM. Thanks!

**Review by zhiics — APPROVED** [review]

**Comment by zhiics** [comment]
Thanks @gussmith23 @comaniac
